azurelinuxagent/common/protocol/wire.py

```diff
@@ -779,17 +779,17 @@ def update_goal_state(self, forced=False, max_retry=3):
 
                 return
 
+            except IOError as e:
+                logger.warn("IOError processing goal state, retrying [{0}]", ustr(e))
```

Comment:
IOError exception object has a strerror member which is probably more appropriate for error messages, instead of converting the exception object to a string.